File: template/php/demo/pic_upload.php
<?php

namespace App;

use BotFaas\QQGuild\DTO\Message;
use BotFaas\QQGuild\DTO\MessageToCreate;
use BotFaas\QQGuild\OpenAPI as QQOpenAPI;
use Pkg\OpenAPI;

/**
 * Class Handler
 * 请将"bot.icodef.com/api/v1/open/guild/pic"加入qq机器人管理的消息URL配置
 * @package App
 */
class Handler
{

    public QQOpenAPI $openapi;

    public function __construct()
    {
        $this->openapi = OpenAPI::OpenAPI();
    }

    public function handle(Message $msg)
    {
        $resp = OpenAPI::UploadPic(new \CURLFile("https://bbs.tampermonkey.net.cn/uc_server/avatar.php?uid=4&size=big&ts=1", "image/png", "test.png"));
        if ($resp['code']) {
            return $this->openapi->MessageAPI()->PostMessage($msg->channel_id, new MessageToCreate([
                'content' => "<@!{$msg->author->id}图片发送失败: " . $resp['msg']
            ]));
        }
        return $this->openapi->MessageAPI()->PostMessage($msg->channel_id, new MessageToCreate([
            'image' => $resp['data']['url'],
            'msg_id' => $msg->id,
        ]));
    }
}

File: template/go/demo/pic_upload/pic_upload.go
package function

import (
	"bytes"
	"context"
	"fmt"
	"handler/pkg/openapi"
	"io"
	"net/http"

	"github.com/tencent-connect/botgo/dto"
)

var api = openapi.OpenAPI()

// Handle a qqbot request
// 请将"bot.icodef.com/api/v1/open/guild/pic"加入qq机器人管理的消息URL配置
func Handle(msg *dto.WSATMessageData) (*dto.Message, error) {
	resp, err := http.Get("https://bbs.tampermonkey.net.cn/uc_server/avatar.php?uid=4&size=big&ts=1")
	if err != nil {
		return api.PostMessage(context.Background(), msg.ChannelID, &dto.MessageToCreate{
			Content: fmt.Sprintf("<@!%s>图片发送失败: %v", msg.Author.ID, err),
			MsgID:   msg.ID,
		})
	}
	buf := bytes.NewBuffer(nil)
	defer resp.Body.Close()
	if _, err := io.Copy(buf, resp.Body); err != nil {
		return api.PostMessage(context.Background(), msg.ChannelID, &dto.MessageToCreate{
			Content: fmt.Sprintf("<@!%s>图片发送失败: %v", msg.Author.ID, err),
			MsgID:   msg.ID,
		})
	}
	pic, err := openapi.UploadPic("test.png", buf)
	if err != nil {
		return api.PostMessage(context.Background(), msg.ChannelID, &dto.MessageToCreate{
			Content: fmt.Sprintf("<@!%s>图片发送失败: %v", msg.Author.ID, err),
			MsgID:   msg.ID,
		})
	}
	if pic.Code == 0 {
		return api.PostMessage(context.Background(), msg.ChannelID, &dto.MessageToCreate{
			Image: pic.Data.Url,
			MsgID: msg.ID,
		})
	}
	return api.PostMessage(context.Background(), msg.ChannelID, &dto.MessageToCreate{
		Content: fmt.Sprintf("<@!%s>图片发送失败: %v", msg.Author.ID, pic.Msg),
		MsgID:   msg.ID,
	})
}
